cmd/fleetctl/main.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/kit/logutil"
+	"github.com/kolide/kit/version"
+)
+
+func runNoop(args []string) error {
+	fmt.Printf("%+v\n", args)
+	return nil
+}
+
+func main() {
+	if len(os.Args) == 2 {
+		if os.Args[1] == "version" {
+			version.PrintFull()
+			os.Exit(0)
+		}
+	}
+
+	logger := level.NewFilter(log.NewJSONLogger(os.Stderr), level.AllowDebug())
+	logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+	logger = log.With(logger, "caller", log.DefaultCaller)
+
+	commands := commandMap{
```

Comment:
A few notes notes about the command list: 
a) It's missing `GET` features. We need to be able to export a pack, which was a big motivation for the original request. 

b) I think the list is very ambitious. We should trim it down to a more basic list of commands we plan to implement and go from there. 

c) There needs to be something to configure the local keys. `fleetctl login` or `fleetctl config set` 

d) I think the operations should be plural, similar to the REST endpoints. That way we can leverage the list methods with filters to get one or more result. For example:
```fleetctl get queries -packs=packs,scheduled,in```

Reply:
A) I intentionally did not include `get` commands. My intention was that if you wanted to "get" a pack or query text, you should just cat the file (which is currently TBD). 

B) Do you see anything that has the potential to be removed / is less MVP than the others?

C) Agreed.

D) Well, there is no `fleetctl get`, so let's ignore that. Everything else is already plural.

Reply:
I don't understand what you mean by intentionally excluding GET. Can you explain how you query current queries/packs/users etc?

Reply:
All of these data model objects are in source controlled files. The fleetcl command merely allows you to sync the changes you made to these files to the server. If you want to "get" a pack on the command line, cat the file.

Reply:
I'm intrigued. This is a big departure from the current model, but I want to hear more of this in an RFC. 

Are you brainstorming a file format? 

Also even though kubernetes works in a similar way, you can still query the current state and dump it back. One of the reasons is that a user can modify a local file and apply it, but not check it in source control. There's no way to diff them if you can't query what's available server side.

Reply:
I could imagine `fleetctl sync` which would download current state and allow you to `git diff` to observe differences.

I have some ideas on file format, I need to write them up.

Reply:
Maybe `fleetctl pull` 🤔